src/main.rs

```diff
@@ -148,6 +121,43 @@ fn main() -> ! {
     let mut led_color = (0u8, 30u8, 255u8);
     let mut led_pulse = false;
 
+    // Set up USB communication.
+    // First we set the D+ pin low for 100ms to simulate a USB
+    // reset condition. This ensures more stable operation when
+    // booting up after a USB DFU firmware update. Without this,
+    // the USB serial device sometimes doesn't show on the host OS
+    // after booting up.
+    let mut delay = hal::delay::Delay::new(cp.SYST, clocks);
+    let mut usb_pin_d_plus = gpioa.pa12.into_push_pull_output();
+    usb_pin_d_plus.set_low().unwrap();
```

Comment:
where are you resetting back? is it done by `UsbDeviceBuilder` later in the code?

Reply:
Yep, once we pass ownership of this pin to the USB structs, they will take care of setting the pins to the proper state for communication.